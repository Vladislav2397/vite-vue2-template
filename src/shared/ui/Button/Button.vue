<template>
    <button :type="type" @click="$emit('click')"><slot /></button>
</template>

<script lang="ts">
import { defineComponent } from 'vue'

type ButtonProps = {
    type: 'button' | 'submit'
}

export default defineComponent({
    name: 'AppButton',
    props: {
        type: {
            type: String as PropType<ButtonProps['type']>,
            default: 'button',
        },
    },
})
</script>

<style scoped>
button {
    border-radius: 8px;
    border: 1px solid transparent;
    padding: 0.6em 1.2em;
    font-size: 1em;
    font-weight: 500;
    font-family: inherit;
    background-color: #1a1a1a;
    cursor: pointer;
    transition: border-color 0.25s;
}
button:hover {
    border-color: #646cff;
}
button:focus,
button:focus-visible {
    outline: 4px auto -webkit-focus-ring-color;
}
</style>
