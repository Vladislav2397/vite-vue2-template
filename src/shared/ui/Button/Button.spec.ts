import { describe, expect, test } from 'vitest'
import { fireEvent, render } from '@testing-library/vue'

import Button from './Button.vue'

describe('Button.spec', () => {
    test('Snapshot', () => {
        const { html } = render(Button, {
            slots: {
                default: 'text',
            },
        })

        expect(html()).toMatchSnapshot()
    })

    test('emit click when clicked by button', async () => {
        const { emitted, getByRole } = render(Button, {
            props: {
                role: 'button',
            },
            slots: {
                default: 'text',
            },
        })

        await fireEvent.click(getByRole('button'))

        expect(emitted()).toHaveProperty('click')
    })
})
