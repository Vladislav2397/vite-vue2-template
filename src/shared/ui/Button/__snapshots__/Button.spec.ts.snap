// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button.spec > Snapshot 1`] = `"<button data-v-053f9315="" type="button">text</button>"`;
